### PR TITLE
WFCORE-316: --cached-dc additional fixes

### PR DESCRIPTION
**Pull request by luck3y**
- keep the remote file repository so it's available when the DC becomes available again
 - parse domain.cached-remote.xml to a temporary file until boot is successful to prevent any boot errors leaving it in a clobbered state
 - if any remote content is unavailable from an offline DC during boot, and not present in the local repository, abort the boot.

### CODE REVIEW

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

File: host-controller/src/main/java/org/jboss/as/host/controller/HostControllerConfigurationPersister.java [hunk 1]
```diff
@@ -82,8 +82,7 @@ public void initializeDomainConfigurationPersister(boolean slave) {
         ConfigurationFile domainConfigurationFile = null;
         if (slave) {
             // in either case of --backup and/or --cached-dc, we persist with the same persister
-            if (environment.isBackupDomainFiles() || environment.isUseCachedDc()) {
-                domainConfigurationFile = getBackupDomainConfigurationFile();
```
Comment (bstansberry) [inline host-controller/src/main/java/org/jboss/as/host/controller/HostControllerConfigurationPersister.java]: Why is this removed? AFAICT the only effect of this is null now gets passed to this below:

```
        // Store this back to environment so mgmt api that exposes it can still work
        environment.setDomainConfigurationFile(domainConfigurationFile);
```

Looking at the impact of that being null, it would only be to disable output from HostEnvironmentResourceDefinition.HostEnvironmentReadHandler. The other read of that value in HostProcessReloadHandler won't happen on a slave, so what we pass in here doesn't matter.

**Comment by bstansberry** [comment]
@luck3y small question above aside, looks good. Ping @ctomc about the compiler failures on windows; we're seeing those on a lot of CI runs. I assume there's some environmental problem on the teamcity servers.

**Comment by luck3y** [comment]
@bstansberry Hmm, that's odd I thought I'd checked what that was used for. Thanks for the catch :)

**Comment by bstansberry** [comment]
@luck3y Please double check my analysis of the code for that point. I could have missed something.

**Comment by luck3y** [comment]
@bstansberry, Removing that wasn't intentional on my part, you're right about:

`// Store this back to environment so mgmt api that exposes it can still work
        environment.setDomainConfigurationFile(domainConfigurationFile);`

Though. I'll retest this to be sure it still works the same.

**Comment by luck3y** [comment]
@bstansberry Can you drop a fixme on this, and I'll ping you or push after I've restested?

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
@bstansberry Windows CI should be fixed, and I've retested that change, and it's working fine, so when the tests pass, should be gtg.

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
Retest this please

**Comment by luck3y** [comment]
@bstansberry this should be all set now
